lib/commands/general.js

```diff
@@ -21,56 +21,42 @@ commands.active = async function active () {
  * Close app (simulate device home button). It is possible to restore
  * the app after the timeout or keep it minimized based on the parameter value.
  *
- * Possible values for `duration`:
- * - any positive number of seconds: come back after X seconds, show deprecation warning
- * - any negative number of seconds: never come back, show deprecation warning
- * - undefined: come back after the default timeout (defined by WDA), show deprecation warning. After deprecation: never come back
+ * @param {?number|Object} seconds
+ * - any positive number of seconds: come back after X seconds
+ * - any negative number of seconds or zero: never come back
+ * - undefined/null: never come back
  * - {timeout: 5000}: come back after 5 seconds
  * - {timeout: null}, {timeout: -2}: never come back
  */
-commands.background = async function background (duration) {
-  const homescreenEndpoint = '/wda/homescreen';
-  const deactivateAppEndpoint = '/wda/deactivateApp';
+commands.background = async function background (seconds) {
+  const homescreen = '/wda/homescreen';
+  const deactivateApp = '/wda/deactivateApp';
+
   let endpoint;
-  let params;
-  if (_.isUndefined(duration)) {
-    // TODO: Replace the block after deprecated stuff is removed
-    // endpoint = homescreenEndpoint;
-    log.warn('commands.background: Application under test will never be restored in the future if no duration is provided. ' +
-             'See https://github.com/appium/appium/issues/7741');
-    endpoint = deactivateAppEndpoint;
-    params = {};
-  } else if (_.isNumber(duration)) {
-    // TODO: deprecate this case
-    log.warn('commands.background: Passing numbers to \'duration\' argument is deprecated. ' +
-             'See https://github.com/appium/appium/issues/7741');
-    if (duration >= 0) {
-      params = {duration};
-      endpoint = deactivateAppEndpoint;
-    } else {
-      endpoint = homescreenEndpoint;
-    }
-  } else if (_.isPlainObject(duration)) {
-    if (_.has(duration, 'timeout')) {
-      if (duration.timeout === null) {
-        endpoint = homescreenEndpoint;
-      } else if (_.isNumber(duration.timeout)) {
-        if (duration.timeout >= 0) {
-          params = {duration: duration.timeout / 1000.0};
-          endpoint = deactivateAppEndpoint;
-        } else {
-          endpoint = homescreenEndpoint;
-        }
+  let params = {};
+  const selectEndpoint = (timeoutSeconds) => {
+    if (!util.hasValue(timeoutSeconds)) {
+      endpoint = homescreen;
+    } if (!isNaN(timeoutSeconds)) {
```

Comment:
Not sure what this line is doing...

Reply:
it check if timeoutSeconds is some random string or other object, which cannot be parsed to a number

Reply:
Why is it on that line?

Reply:
where do you want to see it?

Reply:
there was missing `else`. Should look better now

Reply:
Ah! That makes sense, then. Thanks.